.github/workflows/sonar-check.yml

```
@@ -28,7 +28,7 @@ concurrency:
 
 jobs:
   build:
-    if: github.repository == 'apache/cloudstack'
+    if: github.repository == 'apache/cloudstack' && github.event.pull_request.head.repo.full_name == github.repository
```

```suggestion
    if: github.event.pull_request.head.repo.full_name == github.repository
```
i.e. people may run sonar on their own forks?

---

if so, users need to configure sonar token in github project.

For example, if someone creates a PR in shapeblue (forked) repositories, currently the sonar check is skipped. 
But with this change, sonar check will not be skipped, but fail (as sonar token does not exist in project settings).

---

but only if they create a PR from their own repo to their own repo, so they have control on whether they create the token or not.

---

yes. so between these 2 options with forks
- enable PR sonar check by default. users need to disable it by code changes if they do not have sonar token.
- disable PR sonar check by default. users need to enable it by code changes if they have sonar token.

the PR sonar check was disabled on forks by #7199 
we can change the behaviour if needed

---

but in both cases the code may not be merged, inconvenient :(

---

Right...
Let's assume that most users do not have sonar token, and run sonar check only on apache/cloudstack repository. Make sense ?